Solved #1 - Initialized monorepo

supunTE opened the pull request.

supunTE committed:
Initial commit

supunTE committed:
feat: add server app

Add server app template

Related: Issue#1

supunTE committed:
feat: install husky

Install Husky npm package.

Related: Issue#1

supunTE committed:
feat: initialize models

Initialize models and add templates.

Related: Issue#1

supunTE committed:
feat: add lint-staged

Add lint-staged with prettier.

Related: Issue#1

supunTE committed:
Merge branch 'the-ai-team:main' into main

supunTE committed:
feat: make repo commitzen friendly

install commitzen and make repo commitzen friendly

re #1

pasinducw commented on .editorconfig:
```diff
@@ -0,0 +1,13 @@
+
+[*]
+charset = utf-8
+indent_style = space
```
this does not match with the rules set in `prettierrc.json`

supunTE replied:
fixed

related commit - https://github.com/the-ai-team/razor/commit/1c04b19bd37ac93a17e362f3e71f53c4bf7c2bf3

pasinducw commented on .eslintrc.json:
```diff
@@ -0,0 +1,42 @@
+{
```
please find out if and why the `eslintrc.json` is needed.
- Do we need to use both prettier and eslint?
- What are the things that each of them do?

supunTE replied:
Decided to use only one formatter: prettier.
We don't need eslint as we debug locally and run checks on GitHub CI
related commit - 1c04b19bd37ac93a17e362f3e71f53c4bf7c2bf3

pasinducw replied:
I don't think that ditching eslint is going to be a good idea.
See: https://prettier.io/docs/en/comparison.html

supunTE replied:
Eslint re-implemented.

related commit - 87d08ce15f1ce2ec15951cc6b82016cbd852decc

pasinducw commented on .husky/pre-commit-msg:
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+exec < /dev/tty && npx cz --hook || true
```
Please add a comment, specifying what this line does

supunTE replied:
Comments added

related commit - b57fef5f442d85fd03e9ab36cc91b055d10b1ae1

supunTE replied:
Comments added

related commit - b57fef5f442d85fd03e9ab36cc91b055d10b1ae1

pasinducw commented on .husky/pre-commit:
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npx --no-install lint-staged
```
Please add a comment, specifying what this line does

supunTE replied:
Comments added

related commit - b57fef5f442d85fd03e9ab36cc91b055d10b1ae1

pasinducw commented on .prettierignore:
```diff
@@ -0,0 +1,6 @@
+/dist
+/coverage
+package-lock.json
+node_modules
```
insert newline at the end of file

supunTE replied:
newline added

related commit - a2757e71aa5bff388d5fbedca9e60fb31ed95a15

pasinducw commented on README.md:
```diff
@@ -0,0 +1,94 @@
+[![Commitizen friendly](https://img.shields.io/badge/commitizen-friendly-brightgreen.svg)](http://commitizen.github.io/cz-cli/)
+
+# Razor
```
Can we clean up this file?
- Keep the relevant commands for this project

supunTE replied:
Readme.md cleaned

related commit - df2a47f66abcd4cec7b23e6bdd96c4fb5de83594

pasinducw commented on apps/client-e2e/cypress.config.ts:
```diff
@@ -0,0 +1,6 @@
+import { defineConfig } from 'cypress';
```
Please find out more about `cypress`.
No need to make any changes to this file.

supunTE replied:
Added to things to learn list

pasinducw commented on apps/client-e2e/tsconfig.json:
```diff
@@ -0,0 +1,10 @@
+{
```
find out what the `tsconfig` do.
No need to make any changes to this file

supunTE replied:
Added to things to learn list

pasinducw commented on apps/client/.browserslistrc:
```diff
@@ -0,0 +1,16 @@
+# This file is used by:
```
Find out what browserslist do
No need to make any changes to this file

supunTE replied:
Added to "things to learn list"

pasinducw commented on apps/client/jest.config.ts:
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable */
```
Find out what `jest` do
No need to change anything here

supunTE replied:
Added to "things to learn list"

pasinducw commented on apps/client/src/main.tsx:
```diff
@@ -0,0 +1,14 @@
+import App from './app/app';
+
+const root = ReactDOM.createRoot(
+	document.getElementById('root') as HTMLElement,
```
don't use tabs for spacing.
let's do spaces.

supunTE replied:
done

related commit - https://github.com/the-ai-team/razor/commit/1c04b19bd37ac93a17e362f3e71f53c4bf7c2bf3

pasinducw commented on apps/client/src/polyfills.ts:
```diff
@@ -0,0 +1,7 @@
+ *
+ * See: https://github.com/zloirock/core-js#babel
+ */
+import 'core-js/stable';
```
Find out about polyfills

supunTE replied:
Added to "things to learn list"

pasinducw commented on nx.json:
```diff
@@ -0,0 +1,84 @@
+  "$schema": "./node_modules/nx/schemas/nx-schema.json",
+  "npmScope": "razor",
+  "affected": {
+    "defaultBase": "master"
```
main?

supunTE replied:
fixed

related commit - 64e9d7437f30a461f94620182579c790f8ded129

pasinducw commented on package.json:
```diff
@@ -0,0 +1,73 @@
+  },
+  "private": true,
+  "dependencies": {
+    "commitizen": "^4.2.5",
```
this must be a dev dependency

supunTE replied:
Moved

related commit - dfe257ff30a3f79da26444aeb1fa2a7bf2deb312

supunTE committed:
feat(commitzen): make commitzen to reject commits outside terminal

Commitzen rejects git commits from outside of terminal to use a common pattern in commits.

re #1

supunTE committed:
fix(eslint): remove eslint

remove eslint, application now use only prettier

re #1

supunTE committed:
refactor(husky files): add comment messages specifying what scripts do

Add comment messages specifying what pre-commit and prepare-commit-msg scripts do

re #1

supunTE committed:
style(prettierignore): add new line at the end of file

re #1

supunTE committed:
docs(readme.md): clean readme.md

Remove unnecessary lines in readme.md and add a small description

re #1

supunTE committed:
chore(.gitkeep): remove unwanted .gitkeep

re #1

supunTE committed:
fix(nx.json): change default base to main

re #1

supunTE committed:
build(package.json): add commitzen to dev dependencies

Move commitzen to devDependencies from dependencies

re #1

supunTE committed:
feat(eslint): add eslint

Eslint Initialize with rules

BREAKING CHANGE: Add eslint to linter

re #1

supunTE committed:
docs(readme.md): add how to commit

Add how to commit section

re #1
